Tiny bug in utilities.cc?

When compiling with clang, I get the following warning:

~~~C++
[ 73%] Building CXX object source/multigrid/CMakeFiles/obj_multigrid.debug.dir/mg_base.cc.o
/Users/heltai/dealii/source/base/utilities.cc:703:16: warning: reference cannot be bound to
      dereferenced null pointer in well-defined C++ code; comparison may be assumed to always
      evaluate to true [-Wtautological-undefined-compare]
      Assert (&communicator != 0, ExcInternalError());
               ^~~~~~~~~~~~    ~
/Users/heltai/dealii/include/deal.II/base/exceptions.h:295:11: note: expanded from macro 'Assert'
    if (!(cond))                                                              \
          ^
~~~

I don't know what the assert wanted to test...

--- a/source/base/utilities.cc
+++ b/source/base/utilities.cc
@@ -700,13 +700,8 @@ namespace Utilities
 
     void destroy_communicator (Epetra_Comm &communicator)
     {
-      Assert (&communicator != 0, ExcInternalError());
-
-      // save the communicator, reset
-      // the map, and delete the
-      // communicator if this whole
-      // thing was created as an MPI
-      // communicator
+      // save the communicator, reset the map, and delete the communicator if
+      // this whole thing was created as an MPI communicator
 #ifdef DEAL_II_WITH_MPI
       Epetra_MpiComm
       *mpi_comm = dynamic_cast<Epetra_MpiComm *>(&communicator);

